### PR TITLE
[Feature] compile the third-party packages using pic mode

### PR DESCRIPTION
**Pull request by dujl**
Why I'm doing:
Some third-party packages are not compiled in pic mode result in it can not be used in shared library.

What I'm doing:
compile these package using pic mode

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -431,7 +431,7 @@ build_lzo2() {
 build_bzip() {
     check_if_source_exist $BZIP_SOURCE
     cd $TP_SOURCE_DIR/$BZIP_SOURCE
-
+    CFLAGS="${CFLAGS} -fPIC"
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:434]: this won't work.

File: thirdparty/build-thirdparty.sh [hunk 2]
```diff
@@ -334,7 +334,7 @@ build_simdjson() {
     #ref: https://github.com/simdjson/simdjson/blob/master/HACKING.md
     mkdir -p $BUILD_DIR
     cd $BUILD_DIR
-    $CMAKE_CMD -G "${CMAKE_GENERATOR}" -DCMAKE_CXX_FLAGS="-O3" -DCMAKE_C_FLAGS="-O3" -DSIMDJSON_AVX512_ALLOWED=OFF ..
+    $CMAKE_CMD -G "${CMAKE_GENERATOR}" -DCMAKE_CXX_FLAGS="-O3 -fPIC" -DCMAKE_C_FLAGS="-O3 -fPIC" -DSIMDJSON_AVX512_ALLOWED=OFF ..
```
Comment (kevincai) [inline thirdparty/build-thirdparty.sh:337]: check if `-DCMAKE_POSITION_INDEPENDENT_CODE=True` for this case. refer to https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE

File: thirdparty/build-thirdparty.sh (hunk 1 above)
Comment (dujl) [reply thirdparty/build-thirdparty.sh:434]: no need CFLAGS because we have patch the bzip source code.

File: thirdparty/build-thirdparty.sh (hunk 2 above)
Comment (dujl) [reply thirdparty/build-thirdparty.sh:337]: it works

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by kevincai** [comment]
@mergifyio backport branch-3.2 branch-3.1

**Pull request merged**